moesif-servlet/src/main/java/com/moesif/servlet/MoesifFilter.java

```diff
@@ -427,6 +444,143 @@ private EventResponseModel getEventResponseModel(LoggingHttpServletResponseWrapp
     return eventResponseBuilder.build();
   }
 
+  /***
+   * Method to initialize the batch event processor and create jobs for automatic update of
+   * app config and send events in batches.
+   * Batch event processor maintains batch queue and a taskRunner method to
+   * run scheduled task periodically to send events in batches.
+   */
+  public void initBatchProcessorAndStartJobs() {
+
+    // Create event batch processor for queueing and batching the events.
+    this.batchProcessor = new BatchProcessor(this.moesifApi, this.config, this.debug);
+
+    // Initialize the timer tasks - Create scheduled jobs
+    this.scheduleAppConfigJob();
+    this.scheduleBatchEventsJob();
+  }
+
+  /***
+   * Method to stop the scheduled jobs and drain the batch queue.
+   * Batch event processor sends the leftover events present in queue
+   * before stopping the scheduled jobs.
+   */
+  public void drainQueueAndStopJobs() {
+
+    // Cleanup/transfer the leftover queue events, if any before destroying the filter.
+    this.batchProcessor.run();  // it's ok to run in main thread on destroy.
+
+    // Stop the scheduled jobs
+    try {
+      if (debug) {
+        logger.info("Stopping scheduled jobs.");
+      }
+      this.resetJobTimer(this.updateConfigTimer);
+      this.resetJobTimer(this.sendBatchEventTimer);
+    } catch (Exception e) {
+      // ignore the error.
+    }
+  }
+
+  /**
+   * Method to create scheduled job for updating config periodically.
+   */
+  private void scheduleAppConfigJob() {
```

Comment:
Future change: 5 mins is prob fine for appConfig, but might recommend 1 min or 3 mins in future change once this SDK supports governance rules. Another option is to store Etag and schedule as needed (Etag change or 5 mins, whichever soonest)

Reply:
@dgilling For better performance, the code has been changed to async API call in fire and forget mode because volume of such calls will be high. that means we are not waiting for response/Etag. That's why a timer based jobRunner was added to pull/update appConfig periodically.
I will add support in callback handler and see how it performs.

Reply:
Done